src/pybind/mgr/dashboard/controllers/daemon.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+from typing import Optional
+
+from ..exceptions import DashboardException
+from ..security import Scope
+from ..services.exception import handle_orchestrator_error
+from ..services.orchestrator import OrchClient, OrchFeature
+from . import APIDoc, APIRouter, RESTController
+from ._version import APIVersion
+from .orchestrator import raise_if_no_orchestrator
+
+
+@APIRouter('/daemon', Scope.HOSTS)
+@APIDoc("Perform actions on daemons", "Daemon")
+class Daemon(RESTController):
+    @raise_if_no_orchestrator([OrchFeature.DAEMON_ACTION])
+    @handle_orchestrator_error('daemon')
+    @RESTController.MethodMap(version=APIVersion.EXPERIMENTAL)
+    def set(self, daemon_name: str, action: str = '',
+            container_image: Optional[str] = None):
+
+        if action not in ['start', 'stop', 'restart', 'redeploy']:
+            raise DashboardException(
+                code='invalid_daemon_action',
+                msg=f'Daemon action "{action}" is either not valid or not supported.',
+                http_status_code=400)
```

Comment:
This is the default status code of DashboardException, so you can skip this explicit setting if you want.